### PR TITLE
P5-119: Rewrite

### PR DESCRIPTION
**Pull request by Moesgaarda**
I did a rewrite thing, please fix the todos before merging.

### CODE REVIEW

File: report/5.Solution/FromScreenToMotors.tex [hunk 1]
```diff
@@ -1,85 +1,77 @@
 \section{Mapping between computer vision and movement}
 Two functions are needed to translate the screen position into a velocity on the motors.
-First we need to transfer the screen position to the USB transfer, and then that format to a motor velocity.\todo{Should this be velocity or speed?}
+The first task is to translate the screen position into a format usable for the USB transfer, and then transform that format into a motor velocity.\todo{Should this be velocity or speed?}
```
Comment (nillas12) [inline report/5.Solution/FromScreenToMotors.tex]:
```suggestion
The first task is to translate the screen position into a format that is usable for USB transfer, and then transform that format into a motor velocity.\todo{Should this be velocity or speed?}
```

File: report/5.Solution/FromScreenToMotors.tex [hunk 2]
```diff
@@ -1,85 +1,77 @@
...
-In order to translate the position on the screen to a velocity two functions are needed.
-First we need to transfer the screen position to the USB transfer, and then that format it to a motor velocity.
+To translate the position on the screen into a velocity two functions are needed.
+First of all, it is necessary to make a function maps the numbers correctly. \todo{Which numbers precisely?}
 
-We need to make a function on the python part that maps these numbers.
+This function will require a 2d vector named \textit{pos} as input. \todo{what is pos}
 
-This function will take in 'pos` which is going to be a 2d vector.
+The next goal is to take a location on the screen and translate it into a 8-bit value. \todo{why}
```
Comment (nillas12) [inline report/5.Solution/FromScreenToMotors.tex]: Is it "an" 8-bit value?
```suggestion
The next goal is to take a location on the screen and translate it into an 8-bit value. \todo{why} 
```

File: report/5.Solution/FromScreenToMotors.tex [hunk 3]
```diff
@@ -1,85 +1,77 @@
...
-First we need to transfer the screen position to the USB transfer, and then that format it to a motor velocity.
+To translate the position on the screen into a velocity two functions are needed.
+First of all, it is necessary to make a function maps the numbers correctly. \todo{Which numbers precisely?}
 
-We need to make a function on the python part that maps these numbers.
+This function will require a 2d vector named \textit{pos} as input. \todo{what is pos}
 
-This function will take in 'pos` which is going to be a 2d vector.
+The next goal is to take a location on the screen and translate it into a 8-bit value. \todo{why} 
+Due to the resolution of the camera that is used, the range on the screen is $0 - 480$ on one axis and $0 - 640$ on the other.
```
Comment (nillas12) [inline report/5.Solution/FromScreenToMotors.tex]: Maybe specify what axis' they are. X and Y are pretty universal. Alternatively vertically and horizontally.

File: report/5.Solution/FromScreenToMotors.tex (hunk 2 above)
Comment (nillas12) [inline report/5.Solution/FromScreenToMotors.tex]: We translate it to an 8 bit value for the T target location no?

File: report/5.Solution/FromScreenToMotors.tex [hunk 4]
```diff
@@ -1,85 +1,77 @@
...
 
 \subsection{From USB to motors}
-The NXT receives a vector in range (-127 -> 127, -127 -> 127).
-This ought to get mapped to the speed of the motor which is a value of -100 and 100, on each axis.
-Either full speed in the negatie direction or full speed in the positive direction.
-
-
-The output of the USB is an 8 bit int, and the input of the motor is also an int.
-The easiest way to do these calculations are simply mapping the ranges directly onto each other
+The NXT receives a vector in range $(-127 -> 127, -127 -> 127)$. \todo{correct mathematical notation?}
```
Comment (nillas12) [inline report/5.Solution/FromScreenToMotors.tex]: "in the range", no?
```suggestion
The NXT receives a vector in the range $(-127 -> 127, -127 -> 127)$. \todo{correct mathematical notation?}
```

File: report/5.Solution/FromScreenToMotors.tex [hunk 5]
```diff
@@ -1,85 +1,77 @@
...
+Normally, mapping from input to output would be translating the input into a range of $-1$ to $1$, however, this required floating point precision since it would otherwise result in simply getting -1, 0 or 1, so instead the order is changed up a bit:
 
 $$
 speed(pos) = (pos \cdot range)/input_\text{upperbound} + (dir(pos) \cdot lowerbound)
 $$
 
-An issue with direction rises again with direction, bacause of lowerbound.
-This is fixed by using the direction, so we actually uses the bound correctly.
\ No newline at end of file
+An issue with direction rises again with direction, because of lowerbound.
```
Comment (nillas12) [inline report/5.Solution/FromScreenToMotors.tex]: No clue. I think it has something to do with the lowebound -127 giving issues. Ask casper directly

File: report/5.Solution/FromScreenToMotors.tex [hunk 6]
```diff
@@ -1,85 +1,77 @@
...
-The output of the USB is an 8 bit int, and the input of the motor is also an int.
-The easiest way to do these calculations are simply mapping the ranges directly onto each other
+The NXT receives a vector in range $(-127 -> 127, -127 -> 127)$. \todo{correct mathematical notation?}
+This vector should be mapped to the speed of the motor which is a value between $-100$ and $100$ on each axis where the number corrosponds to a percentage-wise speed in either a positive or negative direction.
 
-However we never want full speed (as that is quite extreme) nor do we want 0 speed (as the motors oftentimes won't move before reaching a power of 10-20 due to sheer weight).
-Therefore the actual range is something like -30 and -10 and 10 and 30
+The output of the USB is an 8-bit integer, and the input of the motor is also an int.
+The easiest way to do these calculations is by simply mapping the ranges directly onto one another.
+However, we rarely want full speed, nor do we want 0 speed as the motors oftentimes will not move until reaching a power level of 10 to 20 due to sheer weight of the attached LEGO components.
```
Comment (nillas12) [inline report/5.Solution/FromScreenToMotors.tex]:
```suggestion
However, we rarely want full speed, nor do we want 0 speed as the motors oftentimes will not move until reaching a power level of 10 to 20 due to the sheer weight of the attached LEGO components.
```

File: report/5.Solution/FromScreenToMotors.tex [hunk 7]
```diff
@@ -1,85 +1,77 @@
...
+Therefore. the actual range was decided to be between $-30$ and $-10$ and likewise in the positive direction.
```
Comment (nillas12) [inline report/5.Solution/FromScreenToMotors.tex]:
```suggestion
Therefore, the actual range was decided to be between $-30$ and $-10$ and likewise in the positive direction.
```

File: report/5.Solution/FromScreenToMotors.tex (hunk 2 above)
Comment (C0DK) [reply report/5.Solution/FromScreenToMotors.tex]:
```
        self.endpoint.write(bytes([int(data.location.x) & 0xFF,
                                   int(data.location.y) & 0xFF,
                                   int(data.status.value[0]) & 0xFF,
                                   0]))
```

Teknisk set 2 8bit værdier.

Men jeg forstår ikke lisbys kommentar

**Pull request merged**